tools/shared.py

```diff
@@ -1243,49 +1253,44 @@ class Building:
   @staticmethod
   def get_multiprocessing_pool():
     if not Building.multiprocessing_pool:
-      Building.multiprocessing_pool = Building.create_multiprocessing_pool()
-    return Building.multiprocessing_pool
-
-  @staticmethod
-  def create_multiprocessing_pool():
-    cores = int(os.environ.get('EMCC_CORES') or multiprocessing.cpu_count())
-
-    # If running with one core only, create a mock instance of a pool that does not
-    # actually spawn any new subprocesses. Very useful for internal debugging.
-    if cores == 1:
-      class FakeMultiprocessor:
-        def map(self, func, tasks):
-          results = []
-          for t in tasks:
-            results += [func(t)]
-          return results
-      pool = FakeMultiprocessor()
-    else:
-      child_env = [
+      cores = int(os.environ.get('EMCC_CORES') or multiprocessing.cpu_count())
+
+      # If running with one core only, create a mock instance of a pool that does not
+      # actually spawn any new subprocesses. Very useful for internal debugging.
+      if cores == 1:
+        class FakeMultiprocessor:
+          def map(self, func, tasks):
+            results = []
+            for t in tasks:
+              results += [func(t)]
+            return results
+        Building.multiprocessing_pool = FakeMultiprocessor()
+      else:
+        child_env = [
+          # Multiprocessing pool children must have their current working directory set to a safe path that is guaranteed not to die in between of
+          # executing commands, or otherwise the pool children will have trouble spawning subprocesses of their own.
+          'CWD='+configuration.TEMP_DIR,
```

Comment:
1. spaces before and after `+`, please
2. how about using `path_from_root()` instead of `configuration.TEMP_DIR`? `path_from_root()` gives the dir in which emscripten lives (i.e. where `emcc.py` lives), so it must exist, while the temp dir could, in theory, be something we want to destroy/create (even though we don't currently, i think)